### PR TITLE
[BUG FIX] Fixed bug with BACKEND_URL; Fixed React error in front end (mismatch between server and client)

### PR DESCRIPTION
**Pull request by sidravi1**
Reviewer: @suzinyou 

Estimate: 30mins

----

# Fixes two front end bug

## Ticket

Fixes: [AAQ-153](https://idinsight.atlassian.net/browse/AAQ-153) and [AAQ-154](https://idinsight.atlassian.net/browse/AAQ-154)

## Description, Motivation and Context

BACKEND_URL was not being picked up since it's server side only
For client side to have access to it we need to use `NEXT_PUBLIC_` as a prefex (or `REACT_APP_`)
But this env variable also needs to be available to the docker during the build because that it where it gets inlined into the code.
This means we have to pass it as argument to docker compose

**NOTE**

This only works because `docker compose` by default loads up the `.env` file that is in the same location as the `docker-compose.yml` file. `.env` has two purposes right now - env variable to `docker compose` itself, runtime env variables to the _containers_ that are created by `docker compose`.

We should decouple this in the future to avoid confusion. Raised ticket [AAQ-155](https://idinsight.atlassian.net/browse/AAQ-155) to track this

### Goal

Fix bugs

### Changes

For AAQ-153:
* Prefixed `NEXT_PUBLIC_` to `BACKEND_URL` to make it available in client side code
* Updated admin app Dockerfile to accept this variable as ARG and use it as ENV variable when building the app
* Updated `docker-compose.yml` to pass on ARG to Dockerfile

For AAQ-154:
* Followed recommendation [here](https://dev.to/adrien/creating-a-custom-react-hook-to-get-the-window-s-dimensions-in-next-js-135k) and moved it to a new component

## How Has This Been Tested?

* Run as dev setup
* Run in docker with WRONG `NEXT_PUBLIC_BACKEND_URL` and note error when calling backend
* Run in docker with RIGHT `NEXT_PUBLIC_BACKEND_URL` and note NO error when calling backend

## Checklist:

This checklist is a useful reminder of small things that can easily be forgotten.
Put an `x` in all the items that apply and remove any items that are not relevant to this PR.

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts
- [x] I have updated the automated tests (if applicable)
- [x] I have written [good commit messages][1]
- [x] I have updated the README file (if applicable)
- [x] I have updated affected documentation (if applicable)

[1]: http://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: admin_app/src/app/login/page.tsx [hunk 1]
```diff
@@ -2,7 +2,8 @@
 
 import React, { useState } from "react";
 
-const backendUrl: string = process.env.BACKEND_URL || "http://localhost:8000";
+const backendUrl: string =
+  process.env.NEXT_PUBLIC_BACKEND_URL || "http://localhost:8000";
```
Comment (suzinyou) [inline admin_app/src/app/login/page.tsx:6]: Is this used in the login page?

**Review by suzinyou — COMMENTED** [review]
I'm not able to log in using the same `.env` file (credentials) I used for logging in on the `main` branch :/

**Review by suzinyou — APPROVED** [review]
I'll approve this for now!

File: admin_app/src/app/login/page.tsx (hunk 1 above)
Comment (sidravi1) [reply admin_app/src/app/login/page.tsx:6]: Yes! The login page called the backend to authenticate and generate a token

**Pull request merged**